src/shared/sol-fbp-parser.c

```diff
@@ -255,7 +256,10 @@ parse_inport_stmt(struct sol_fbp_parser *p)
 
     sol_fbp_graph_add_in_port(p->graph, node_idx, port, get_token_position(&p->current_token));
 
-    err = sol_fbp_graph_add_exported_in_port(p->graph, node_idx, port, port_idx, exported_port, get_token_position(&p->current_token), &ep);
+    if ((long)(int)port_idx != port_idx)
+        return false;
```

Comment:
above you did cast to `long int`, although they are identical, keep it consistent choosing one way to do the cast.